### PR TITLE
Cleanup: combine build/provenance binaries and use flags lib

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>
part of https://github.com/asraa/slsa-on-github/issues/32

### CODE REVIEW

**Comment by asraa** [comment]
Need to rebase off #77 and #78

**Review by asraa — COMMENTED** [review]
This is done -- @laurentsimon since sigstore is bringing in relatively high deps, is it worth separating out the provenance binary by using a build tag so the build binary doesn't pull that them in? But at least we're moving closer to a monorepo

**Review by laurentsimon — APPROVED** [review]

File: .github/workflows/builder.yml [hunk 1]
```diff
@@ -7,15 +7,12 @@ env:
...
-  # Provenance generation.
-  PROVENANCE_BINARY: provenance
-  PROVENANCE_DIR: slsa-provenance
   # Builder
   BUILDER_BINARY: builder
   BUILDER_DIR: builder
   # This repo.
   BUILDER_REPOSITORY: asraa/slsa-on-github
-  BUILDER_REF: main
+  BUILDER_REF: cleanup-flags
```
Comment (laurentsimon) [inline .github/workflows/builder.yml:15]: remove after tests

**Review by laurentsimon — APPROVED** [review]

**Comment by laurentsimon** [comment]
> This is done -- @laurentsimon since sigstore is bringing in relatively high deps, is it worth separating out the provenance binary by using a build tag so the build binary doesn't pull that them in? But at least we're moving closer to a monorepo

what do you mean by build tag?

**Comment by asraa** [comment]
> what do you mean by build tag?

If we annotate the provenance with a tag, then we can optionally compile it. e.g if you only want the builder you would not add the tag, but if you do you would build with something like `go build -tags=provenance -o builder`

**Comment by asraa** [comment]
We can open an issue if we want to separate the builds with a tag in the next repo. For now at least the slsa builder is unified so we don't have two go modules when we move repos

**Comment by laurentsimon** [comment]
> We can open an issue if we want to separate the builds with a tag in the next repo. For now at least the slsa builder is unified so we don't have two go modules when we move repos

sgtm

**Pull request merged**